streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```
@@ -73,203 +55,35 @@ public static boolean isValidCleanupPolicy(final String cleanupPolicy) {
-
-    @SuppressWarnings("unchecked")
-    private Map<Integer, List<Integer>> getTopicMetadata(String topic) {
-        String data = zkClient.readData(ZK_TOPIC_PATH + "/" + topic, true);
-
-        if (data == null) return null;
-
-        try {
-            ObjectMapper mapper = new ObjectMapper();
-
-            Map<String, Object> dataMap = mapper.readValue(data, new TypeReference<Map<String, Object>>() {
-
-            });
-
-            Map<Integer, List<Integer>> partitions = (Map<Integer, List<Integer>>) dataMap.get("partitions");
 
-            log.debug("Read partitions {} for topic {} from ZK in partition assignor.", partitions, topic);
-
-            return partitions;
-        } catch (IOException e) {
-            throw new StreamsException("Error while reading topic metadata from ZK for internal topic " + topic, e);
-        }
-    }
-
-    private void createTopic(InternalTopicConfig topic, int numPartitions, int replicationFactor) throws ZkNodeExistsException {
-        log.debug("Creating topic {} with {} partitions from ZK in partition assignor.", topic.name(), numPartitions);
-        ObjectMapper mapper = new ObjectMapper();
-        List<Integer> brokers = getBrokers();
-        int numBrokers = brokers.size();
-        if (numBrokers < replicationFactor) {
-            log.warn("Not enough brokers found. The replication factor is reduced from " + replicationFactor + " to " +  numBrokers);
-            replicationFactor = numBrokers;
-        }
-
-        Map<Integer, List<Integer>> assignment = new HashMap<>();
-
-        for (int i = 0; i < numPartitions; i++) {
-            ArrayList<Integer> brokerList = new ArrayList<>();
-            for (int r = 0; r < replicationFactor; r++) {
-                int shift = r * numBrokers / replicationFactor;
-                brokerList.add(brokers.get((i + shift) % numBrokers));
-            }
-            assignment.put(i, brokerList);
+        if (!streamsKafkaClient.topicExists(topic.name())) {
```

If the topic exists we need to make sure the number of partitions is the same. If it isn't the same then we should throw an exception.

---

Added the check for the number of partitions.